### ISSUE TITLE
`{:bad_generator, 252}` when running README example

### ISSUE THREAD

**Issue by formido** [breakroom/multipart#4]
Love the idea of this project since creating complex multipart in HTTPoison has proven to be such a pain. 

But tried the homepage example and got:

```
** (ErlangError) Erlang error: {:bad_generator, 252}
    (mint 1.3.0) lib/mint/http1/request.ex:84: Mint.HTTP1.Request."-validate_header_value!/2-lc$^0/1-0-"/3
    (mint 1.3.0) lib/mint/http1/request.ex:84: Mint.HTTP1.Request.validate_header_value!/2
    (mint 1.3.0) lib/mint/http1/request.ex:27: anonymous fn/2 in Mint.HTTP1.Request.encode_headers/1
    (elixir 1.13.0) lib/enum.ex:2396: Enum."-reduce/3-lists^foldl/2-0-"/3
    (mint 1.3.0) lib/mint/http1/request.ex:9: Mint.HTTP1.Request.encode/4
    (mint 1.3.0) lib/mint/http1.ex:242: Mint.HTTP1.request/5
    (finch 0.10.2) lib/finch/http1/conn.ex:114: Finch.Conn.request/6
    (finch 0.10.2) lib/finch/http1/pool.ex:46: anonymous fn/8 in Finch.HTTP1.Pool.request/5
    (nimble_pool 0.2.4) lib/nimble_pool.ex:266: NimblePool.checkout!/4
    (finch 0.10.2) lib/finch/http1/pool.ex:39: Finch.HTTP1.Pool.request/5
    (finch 0.10.2) lib/finch.ex:275: Finch.request/3
```

This is with Finch 0.10.2 and my details:

```
MacBook-Pro:multipart michael$ elixir --version
Erlang/OTP 24 [erts-12.1.5] [source] [64-bit] [smp:10:10] [ds:10:10:10] [async-threads:1] [dtrace]

Elixir 1.13.0 (compiled with Erlang/OTP 24)
```

**Comment by formido** [comment]
OK, figured it out:

Mint is [validating the header values](https://github.com/elixir-mint/mint/blob/b706ba33dbabd5b4e137caafc1cd01c7228e0c70/lib/mint/http1/request.ex#L82-L91) like this:

```elixir
defp validate_header_value!(name, value) do
  _ =
    for <<char <- value>> do
      unless is_vchar(char) or char in '\s\t' do
        throw({:mint, {:invalid_header_value, name, value}})
      end
    end


  :ok
end
```

...but `content_length` in the README example is an integer.

**Comment by tomtaylor** [comment]
Thanks - fixed in a932e19a96a409d5b51dbf1ebba4a0ff15b1009e.
